### PR TITLE
Audience Evaluation Logging

### PR DESCRIPTION
**Pull request by yasirfolio3**

### CODE REVIEW

**Comment by yasirfolio3** [comment]
Closing this PR, Changes now moved to:
https://github.com/optimizely/objective-c-sdk/pull/362

**Review by nchilada — COMMENTED** [review]
Nice work @yasirfolio3 @msohailhussain! I'm so sorry I didn't get to this in time. I just skimmed this now and have a couple suggestions that I hope we can follow up on, if they make sense.

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYNSObject+Validation.m [hunk 1]
```diff
@@ -64,6 +64,50 @@ - (NSString *)getStringOrEmpty {
...
+        if ([self isKindOfClass:[NSDictionary class]] && ((NSDictionary *)self).count > 0) {
+            NSError *error = nil;
+            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:(NSDictionary *)self options:NSJSONWritingPrettyPrinted error:&error];
+            if (error == nil) {
+                string = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+            }
+        }
+    }
+    return string;
+}
```
[the remark this answers is not in the document]
Comment (yasirfolio3) [reply OptimizelySDKCore/OptimizelySDKCore/OPTLYNSObject+Validation.m:93]: getJSONArrayStringOrEmpty is in use while getJSONDictionaryStringOrEmpty has been removed. This PR was closed and #362 was made for Audience logging. i'll review and make the changes you suggested in the below branch and make a separate PR for it if they weren't already included in #362 .
https://github.com/optimizely/objective-c-sdk/tree/yasir/audience-eval-logging-2

**Comment by yasirfolio3** [comment]
@nchilada Changes made in #372 , please review.

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYNSObject+Validation.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (nchilada) [reply OptimizelySDKCore/OptimizelySDKCore/OPTLYNSObject+Validation.m:93]: Okay, thanks!!

**Pull request closed without merging**